### PR TITLE
Github Deployment: Fetch log details when a long entry show more is clicked

### PR DESCRIPTION
**Pull request by vykes-mac**
Closes https://github.com/Automattic/dotcom-forge/issues/6014. Closes https://github.com/Automattic/dotcom-forge/issues/5852.

## Proposed Changes

* fetch log details from the new endpoint when the `show more` is expanded

## Testing Instructions

* go to deployment logs expand the logs and click `show more`

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/github-deployments/deployment-run-logs/deployment-run-logs.tsx [hunk 1]
```diff
@@ -1,75 +1,103 @@
...
+	const hasDetail = !! commandIdentifier;
+
+	const { data: logDetail, isLoading } = useCodeDeploymentsRunLogDetailQuery(
+		siteId,
+		deployment.id,
+		run.id,
+		commandIdentifier ?? null,
+		{
+			enabled: detailExpanded && hasDetail,
+			refetchInterval: 5000,
```
Comment (zaguiini) [inline client/my-sites/github-deployments/deployment-run-logs/deployment-run-logs.tsx]: I don't think we need to refetch this endpoint. It's not like the logs for an identifier are going to change.

Comment (zaguiini) [reply client/my-sites/github-deployments/deployment-run-logs/deployment-run-logs.tsx]: @jeroenpf is there any necessity to poll this endpoint? Just to confirm.

Comment (jeroenpf) [reply client/my-sites/github-deployments/deployment-run-logs/deployment-run-logs.tsx]: No, it won't change after its been logged so we don't need to poll it.

**Comment by vykes-mac** [comment]
Tested well with the wpcomsh changes!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11602081

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @vykes-mac for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11602081).

**Pull request merged**